app/javascript/components/mentoring/MentorDiscussion.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from 'react'
+import { MentoringPanelList } from './mentor-discussion/MentoringPanelList'
+import { IterationsList } from './mentor-discussion/IterationsList'
+import { BackButton } from './mentor-discussion/BackButton'
+import { SolutionInfo } from './mentor-discussion/SolutionInfo'
+
+type Links = {
+  scratchpad: string
+  exercise: string
+}
+
+export type Iteration = {
+  idx: number
+  numComments: number
+  unread: boolean
+  links: {
+    posts: string
+  }
+}
+
+export type Student = {
+  avatarUrl: string
+  handle: string
+}
+
+export type Track = {
+  title: string
+  iconUrl: string
+}
+
+export type Exercise = {
+  title: string
+}
+
+export const MentorDiscussion = ({
+  student,
+  track,
+  exercise,
+  links,
+  discussionId,
+  iterations,
+}: {
+  student: Student
+  track: Track
+  exercise: Exercise
+  links: Links
+  discussionId: number
+  iterations: Iteration[]
+}): JSX.Element => {
+  const [currentIteration, setCurrentIteration] = useState(
+    iterations[iterations.length - 1]
```

Comment:
Is there any scenario where `iterations` can be empty? Put another way: do we need to guard against the iterations possibly being empty?

Reply:
I don't think so. You can't have a mentor discussion without submitting an iteration. But yes, I'll throw a clearer error when that happens.